src/forms/timeInput.test.jsx

```diff
@@ -1,49 +1,250 @@
+	HOURS,
+	MINUTES,
+	MERIDIAN,
+	getTimeParts
+} from './TimeInput';
+
 
 describe('TimeInput', function() {
 
 	let component,
+		component12Hr,
+		hoursInputEl,
+		minutesInputEl,
 		inputEl,
-		onChangeSpy;
+		onChangeSpy,
+		onNumberChangeSpy,
+		onMeridianChangeSpy,
+		onTimeInputChangeSpy;
 
 	const onChangePropMock = jest.fn(),
-		newTime = '23:00';
+		newTime = '23:00',
+		newHour = '23',
+		overMax = '666',
+		underMax = '-666';
 
 	const props = {
 		name: 'time',
-		value: '22:00',
+		value: '22:12',
 		onChange: onChangePropMock,
 		required: true
 	};
 
-	beforeEach(() => {
-		onChangeSpy = jest.spyOn(TimeInput.prototype, 'onChange');
-		component = shallow(<TimeInput {...props} />);
-		inputEl = component.find('input');
-	});
+	describe('TimeInput, basic functionality', () => {
 
-	afterEach(() => {
-		component = null;
-		inputEl = null;
-		onChangeSpy.mockClear();
-	});
+		describe('getTimeParts', () => {
+			it('returns an object if no parts specified', () => {
+				const result = getTimeParts('13:00');
+				const expected = {
+					[HOURS]: '13',
+					[MINUTES]: '00',
+					[MERIDIAN]: 'PM'
+				};
+				expect(result).toEqual(expected);
+			});
+			it('returns a part if part specified', () => {
+				const result = getTimeParts('13:00', HOURS);
+				expect(result).toEqual('13');
+			});
+		});
 
-	it('renders a time html input with expected props', function() {
-		expect(component).toMatchSnapshot();
+		describe('parseValueForState', () => {
+			it('takes a value string for 24Hr time and returns an object with hours, min, meridian, value', () => {
+				const value = '13:00',
+				is24Hr = true;
+				const result = TimeInput.prototype.parseValueIntoState(value, is24Hr);
+				const expected = {
+					[HOURS]: '13',
+					[MINUTES]: '00',
+					[MERIDIAN]: false,
+					value
+				};
+				expect(result).toEqual(expected);
+			});
+			it('takes a value string, is24hr as false, and returns an object with hours, min, meridian, value', () => {
+				const value = '13:15',
+					is24Hr = false;
+				const result = TimeInput.prototype.parseValueIntoState(value, is24Hr);
+				const expected = {
+					[HOURS]: '01',
+					[MINUTES]: '15',
+					[MERIDIAN]: 'PM',
+					value
+				};
+				expect(result).toEqual(expected);
+			});
+		});
+
+		describe('constrainValue', () => {
+			it('returns the same value if no min and max', () => {
+				const value = '10';
+				const result = TimeInput.prototype.constrainValue(undefined, undefined, value);
+				expect(result).toEqual(parseInt(value));
+			});
+			it('returns the constrained value if min', () => {
+				const value = '10';
+				const min = 20;
+				const result = TimeInput.prototype.constrainValue(min, undefined, value);
+				expect(result).toEqual(min);
+			});
+			it('returns the constrained value if max', () => {
+				const value = '10';
+				const max = 5;
+				const result = TimeInput.prototype.constrainValue(undefined, max, value);
+				expect(result).toEqual(max);
+			});
```

Comment:
probably good to add a test for the 'no change' case when constraints _are_  provided 
- e.g. 'returns unchanged value if within constraints'